service/worker/deletenamespace/reclaimresources/workflow.go

```diff
@@ -0,0 +1,170 @@
+	WorkflowName = "temporal-sys-reclaim-namespace-resources-workflow"
+
+	checkMaxAttempts = 10
+)
+
+type (
+	ReclaimResourcesParams struct {
+		deleteexecutions.DeleteExecutionsParams
+	}
+
+	ReclaimResourcesResult struct {
+		SuccessCount int
+		ErrorCount   int
+	}
+)
+
+var (
+	retryPolicy = &temporal.RetryPolicy{
+		InitialInterval: 1 * time.Second,
+		MaximumInterval: 10 * time.Second,
+	}
+
+	localActivityOptions = workflow.LocalActivityOptions{
+		RetryPolicy:            retryPolicy,
+		StartToCloseTimeout:    10 * time.Second,
+		ScheduleToCloseTimeout: 5 * time.Minute,
+	}
+
+	deleteExecutionsWorkflowOptions = workflow.ChildWorkflowOptions{
+		RetryPolicy:              retryPolicy,
+		WorkflowRunTimeout:       60 * time.Minute,
+		WorkflowExecutionTimeout: 24 * time.Hour,
+	}
+)
+
+func validateParams(params *ReclaimResourcesParams) error {
+	if params.NamespaceID.IsEmpty() {
+		return temporal.NewNonRetryableApplicationError("namespace ID is empty", "", nil)
+	}
+
+	if params.Namespace.IsEmpty() {
+		return temporal.NewNonRetryableApplicationError("namespace is empty", "", nil)
+	}
+
+	return nil
+}
+
+func ReclaimResourcesWorkflow(ctx workflow.Context, params ReclaimResourcesParams) (ReclaimResourcesResult, error) {
+	logger := workflow.GetLogger(ctx)
+	logger.Info("Child workflow started.", tag.WorkflowType(WorkflowName))
+
+	if err := validateParams(&params); err != nil {
+		return ReclaimResourcesResult{}, err
+	}
+
+	var a *Activities
+
+	// Step 1. Delete workflow executions.
+	result, err := deleteWorkflowExecutions(ctx, params, a, logger)
+	if err != nil {
+		return result, err
+	}
+
+	// Step 2. Delete namespace.
+	ctx5 := workflow.WithLocalActivityOptions(ctx, localActivityOptions)
+	err = workflow.ExecuteLocalActivity(ctx5, a.DeleteNamespaceActivity, params.Namespace, params.NamespaceID).Get(ctx, nil)
+	if err != nil {
+		return result, fmt.Errorf("%w: DeleteNamespaceActivity: %v", errors.ErrUnableToExecuteActivity, err)
+	}
+
+	logger.Info("Child workflow finished successfully.", tag.WorkflowType(WorkflowName))
+	return result, nil
+}
+
+func deleteWorkflowExecutions(ctx workflow.Context, params ReclaimResourcesParams, a *Activities, logger log.Logger) (ReclaimResourcesResult, error) {
+	deleteAttempt := int32(1)
+	executionsExist := true
+	var result ReclaimResourcesResult
+	for executionsExist {
+		ctx1 := workflow.WithChildOptions(ctx, deleteExecutionsWorkflowOptions)
+		var der deleteexecutions.DeleteExecutionsResult
+		err := workflow.ExecuteChildWorkflow(ctx1, deleteexecutions.DeleteExecutionsWorkflow, params.DeleteExecutionsParams).Get(ctx, &der)
+
+		if err != nil {
+			logger.Error("Unable to execute child workflow.", tag.WorkflowType(deleteexecutions.WorkflowName), tag.Error(err))
+			return result, fmt.Errorf("%w: %s: %v", errors.ErrUnableToExecuteChildWorkflow, deleteexecutions.WorkflowName, err)
+		}
+		result.SuccessCount += der.SuccessCount
+		result.ErrorCount += der.ErrorCount
```

Comment:
If visibility is lagging behind, you could call DeleteExecutionsWorkflow multiple times and the executions would all be gone already and delete would return NotFound, but the SuccessCount will be double counted. Which might be fine, just saying.